lib/dora-parser/src/parser.rs

```diff
@@ -472,18 +472,23 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
-    fn parse_modifiers(&mut self) -> Result<Modifiers, MsgWithPos> {
+    fn parse_annotations(&mut self) -> Result<Modifiers, MsgWithPos> {
         let mut modifiers = Modifiers::new();
         loop {
-            let modifier = match self.token.kind {
-                TokenKind::Abstract => Modifier::Abstract,
-                TokenKind::Open => Modifier::Open,
-                TokenKind::Override => Modifier::Override,
-                TokenKind::Final => Modifier::Final,
-                TokenKind::Internal => Modifier::Internal,
-                TokenKind::Pub => Modifier::Pub,
-                TokenKind::Static => Modifier::Static,
-                TokenKind::Optimize => Modifier::Optimize,
+            if !self.token.is(TokenKind::At) {
+                break;
+            }
+            self.advance_token()?;
+            let ident = self.expect_identifier()?;
+            let modifier = match self.interner.str(ident).as_str() {
+                "abstract" => Modifier::Abstract,
+                "open" => Modifier::Open,
+                "override" => Modifier::Override,
+                "final" => Modifier::Final,
+                "internal" => Modifier::Internal,
+                "pub" => Modifier::Pub,
+                "static" => Modifier::Static,
+                "optimize" => Modifier::Optimize,
                 _ => {
                     break;
```

Comment:
I think we should throw an error here. We are already parsing `@<identifier>` here, so this should throw an error like `unknown annotation <identifier>`...

Reply:
Will do!

Reply:
Done.